server/game/gameSystems/ReadySystem.ts

```diff
@@ -0,0 +1,45 @@
+import type { AbilityContext } from '../core/ability/AbilityContext';
+import type { Card } from '../core/card/Card';
+import { CardType, EventName, Location, WildcardCardType } from '../core/Constants';
+import * as EnumHelpers from '../core/utils/EnumHelpers';
+import { type ICardTargetSystemProperties, CardTargetSystem } from '../core/gameSystem/CardTargetSystem';
+import { CardWithExhaustProperty } from '../core/card/CardTypes';
+import { IGameSystemProperties } from '../core/gameSystem/GameSystem';
+
+export interface IReadySystemProperties extends ICardTargetSystemProperties {
+    isRegroupPhaseReadyStep?: boolean;
+}
+
+export class ReadySystem extends CardTargetSystem<IReadySystemProperties> {
+    public override readonly name = 'ready';
+    public override readonly eventName = EventName.OnCardExhausted;
+    public override readonly costDescription = 'readying {0}';
+    public override readonly effectDescription = 'ready {0}';
+    protected override readonly targetTypeFilter = [WildcardCardType.Unit, CardType.Leader, CardType.Event];
+    protected override readonly defaultProperties: IReadySystemProperties = {
+        isRegroupPhaseReadyStep: false
+    };
+
+    public eventHandler(event): void {
+        event.card.ready();
+    }
+
+    public override canAffect(card: Card, context: AbilityContext): boolean {
```

Comment:
I think we need a check here eventually for effects that prevent readying like Frozen in Carbonite and No Good to Me Dead. Something like checking checking for a cardCannot be readied ongoing effect?

If we implement that when we add those cards, I'd recommend leaving a TODO here for it